docs/data/netCDF/regularGridExample.rst

```diff
@@ -1,9 +1,21 @@
 .. _regularGridExample:
 
-Regular Grids
--------------
+Regular Grid Example in Python
+==============================
 
-Here we demonstrate how to use *xarray* to apply the CF Conventions to a basic NetCDF file (`simple.nc <https://drive.google.com/file/d/18CFrtFxvHwfL30Xd91TNU38J_z-Kgywm/view?usp=sharing>`_) to define it on a regular, rectilinar grid for ingesting into Vapor.  You can download `this Python script <https://github.com/NCAR/VAPOR/blob/readTheDocs/docs/data/regularToCF.py>`_ that replicates the steps below.
+.. raw:: html
+
+    <iframe width="560" height="315" src="https://www.youtube.com/embed/lYzj3ZQMlBE" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+|
+
+Download this example in python :download:`here <regularToCF.py>`
```

Comment:
You're raising a good point and there is a bit of a story to this.  My updates to fix this problem are in #2912.

In this implementation I had python scripts (like regularToCF.py) that did the job.  I took these and copied them into notebooks in a piece-by-piece fashion.  Users could now download these .py and .ipynb files.

Additionally, I copied and reformatted the code into .rst files that were being built with a module called [jupyter-sphinx](https://blog.jupyter.org/integrating-output-in-documentation-with-jupyter-sphinx-ecf569ddab85).  jupyter-sphinx is a fairly widely adopted module that lets users format .rst files so that they render as if they were run in a jupyter notebook.

But this left us with three copies of code that was doing the same thing, which you've pointed out.

The GeoCat team is using a module called sphinx-gallery that generates .html and .ipynb files from a single .py source file.  It also automatically provides links to download the .ipynb and .py files that are given as examples.  I originally didn't think of our netCDF examples as being a gallery, but it should be with the right sphinx-gallery configuration.

Check out #2912 to see the changes.  I've included links to the current documentation build and the #2912 build so you can compare them side by side.  There is also some structural cleaning in #2912.  This allows us to have only one copy of the logic, and also falls in line with what our teammates at GeoCat are doing.